HealthcareApp/src/Healthcare.Application/Schedules/Queries/AvailableTimeSlots/GetAvailableSlotsQueryHandler.cs

```diff
@@ -0,0 +1,39 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Application.DTOs.Result;
+using Healthcare.Application.DTOs.Schedule;
+using MediatR;
+
+namespace Healthcare.Application.Schedules.Queries.AvailableTimeSlots;
+
+public class
+    GetAvailableSlotsQueryHandler : IRequestHandler<GetAvailableSlotsQuery, Result<Dictionary<string, List<TimeSlotDTO>>>>
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly IScheduleRepository _scheduleRepository;
+
+
+    public GetAvailableSlotsQueryHandler(IUserManagerDecorator<ApplicationUser> userManager,
+        IScheduleRepository scheduleRepository)
+    {
+        _userManager = userManager;
+        _scheduleRepository = scheduleRepository;
+    }
+
+    public async Task<Result<Dictionary<string, List<TimeSlotDTO>>>> Handle(GetAvailableSlotsQuery request,
+        CancellationToken cancellationToken)
+    {
+        var doctor = await _userManager.FindByIdAsync(request.DoctorId);
+        if (doctor == null) return Result<Dictionary<string, List<TimeSlotDTO>>>.Failure("Doctor not found");
```

Comment:
If you make `Failure` method generic, and then move this method from your generic result, you will be able to write here just:
`return Result.Failure("Doctor not found");` and compiler should know what this result is, because you declare it in your method signature.

This is just cosmetical change, not mandatory to fix